packages/cli/src/util/config.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { JSONSchemaType } from "ajv";
+import { promises as fsPromises } from "node:fs";
+
+type AutoVersionType = "git-describe";
+
+interface AutoVersion {
+  type: AutoVersionType;
+  tagPrefix?: string;
+}
```

Comment:
suggestion: Start this off as a type union as tagPrefix may only make sense for git-describe

```suggestion
interface GitDescribeAutoVersionConfig {
  type: "git-describe";
  tagPrefix?: string;
}

type AutoVersionConfig = GitDescribeAutoVersionConfig;
type AutoVersionType = AutoVersionConfig["type"];
```